src/main/scala/shine/DPIA/Compilation/CodeGenerator.scala

```diff
@@ -48,13 +48,17 @@ trait CodeGenerator {
 object CodeGenerator {
   final case class
     Environment(identEnv: immutable.Map[Identifier[_ <: BasePhraseType], C.AST.DeclRef],
+                varEnv: immutable.Map[Identifier[VarType], C.AST.DeclRef],
```

Comment:
I am not 100% sure that it cannot be eliminated, but right now it is required because depending on the context, a managed buffer might be used as a regular acceptor/expression, or as a full variable, and can be represented with an acceptor/expression identifier (input/output) or a variable identifier (temporary). In any case the buffer use should generate the same C identifier.

Reply:
Could we combine this with the `identEnv` map to allow identifiers with a var type to appear in them as well (i.e., relaxing the type of the `identEnv` map)?

Reply:
We have code such as `C.CodeGenerator.applySubstitutions` that relies on the `_ <: BasePhraseType` bound, but I can try to relax it and fix all uses that rely on this bound.

Reply:
I removed `varEnv` altogether, I noticed that it was never actually used for anything anymore (only updated).